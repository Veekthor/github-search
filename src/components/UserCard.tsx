import { IUserCardProps } from "../interfaces";
import { StyledUserCard } from "./styles/Results.styled";
import { IoIosArrowForward } from "react-icons/io";

const UserCard: React.FC<IUserCardProps> = ({ user }) => {
  return (
    <StyledUserCard>
      <div>
        <img
          src={user.avatar_url}
          alt={`${user.login}'s avatar`}
          loading="lazy"
        />
      </div>
      <div>
        <p>{user.login}</p>
        <IoIosArrowForward />
      </div>
    </StyledUserCard>
  );
};

export default UserCard;
