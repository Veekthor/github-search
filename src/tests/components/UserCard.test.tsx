import renderer from "react-test-renderer";
import UserCard from "../../components/UserCard";
import fetchedUsersObj from "../fixtures/data";

describe('UserCard tests', () => {
  it('should match snapshot', () => {
    expect(true).toBe(true);
    const user = fetchedUsersObj.items[0];
    const component = renderer.create(<UserCard user={user} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})