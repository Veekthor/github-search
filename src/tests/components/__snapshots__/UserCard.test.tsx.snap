// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard tests should match snapshot 1`] = `
<div
  className="sc-gTRrQi iWBeRi"
  data-testid="user-card"
>
  <div>
    <img
      alt="Veek's avatar"
      loading="lazy"
      src="https://avatars.githubusercontent.com/u/7710803?v=4"
    />
  </div>
  <div>
    <p>
      Veek
    </p>
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M294.1 256L167 129c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.3 34 0L345 239c9.1 9.1 9.3 23.7.7 33.1L201.1 417c-4.7 4.7-10.9 7-17 7s-12.3-2.3-17-7c-9.4-9.4-9.4-24.6 0-33.9l127-127.1z"
      />
    </svg>
  </div>
</div>
`;
